cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/DefaultKafkaQuotasManager.java

```diff
@@ -0,0 +1,227 @@
+                    if (isNotKafkaPlugin || ops.stream().allMatch(op -> Objects.isNull(op.value()))) {
+                        LOGGER.debugCr(reconciliation, "There are no default user quotas set in Kafka and the Kafka built-in plugin is not configured, skipping the alteration");
+                        return Future.succeededFuture(false);
+                    } else {
+                        // in case that the current quotas are null, but desired quotas contains some non-null values, we should alter the quotas
+                        return Future.succeededFuture(true);
+                    }
+                }
+
+                return Future.succeededFuture(currentAndDesiredQuotasDiffer(currentQuotas, ops));
+            });
+    }
+
+    /**
+     * Method for altering the default Kafka user quotas using Kafka Admin client
+     *
+     * @param reconciliation            Reconciliation marker
+     * @param vertx                     Vert.x instance
+     * @param kafkaAdmin                Kafka Admin object
+     * @param clientQuotaAlteration     Quota alteration operation
+     *
+     * @return  Future after completion of the alter operation
+     */
+    private static Future<Void> alterQuotas(
+        Reconciliation reconciliation,
+        Vertx vertx,
+        Admin kafkaAdmin,
+        ClientQuotaAlteration clientQuotaAlteration
+    ) {
+        LOGGER.infoCr(reconciliation, "Altering default user quotas to: {}", clientQuotaAlteration.toString());
+
+        return VertxUtil
+            .kafkaFutureToVertxFuture(reconciliation, vertx, kafkaAdmin.alterClientQuotas(List.of(clientQuotaAlteration)).values().get(DEFAULT_USER_ENTITY))
+            .map((Void) null);
+    }
+
+    /**
+     * Method that compares the desired quotas, represented by List of {@link ClientQuotaAlteration.Op}, and current quotas
+     * set in Kafka, represented by Map.
+     *
+     * @param desiredQuotas     desired quotas, represented as List of {@link ClientQuotaAlteration.Op}
+     * @param currentQuotas     current quotas configured in Kafka, represented as Map
+     *
+     * @return  boolean result of the comparison. Returns true if both current and desired quotas are same, false otherwise
+     */
+    private static boolean currentAndDesiredQuotasDiffer(Map<String, Double> currentQuotas, List<ClientQuotaAlteration.Op> desiredQuotas) {
+        // desiredQuotas will always contain all quotas keys, because it is filled from `prepareQuotaConfigurationRequest`
+        // that's why we can iterate through the list and be sure that we cover all the quota keys
+        for (ClientQuotaAlteration.Op quota : desiredQuotas) {
+            Double currentValue = currentQuotas.get(quota.key());
+            Double desiredValue = quota.value();
+
+            if (!Objects.equals(currentValue, desiredValue)) {
```

Comment:
What does this do for Double objects? Does it work? Is this used because of null values? I do not see any tests for this.

Reply:
It is used because of the possible `null` values in both. And I tested it + it's tested as part of the `DefaultKafkaQuotasManagerTest`, where I'm checking if the quotas should be altered in case that the config is null, the same fields are filled but with different values, etc.

I can add tests for this method as well.

NOTE: because I'm checking if the `currentQuotas` is `null` or not in the previous steps of the `shouldAlterDefaultQuotasConfig`, I'm not handling the `null` in this compare method (which can be fixed, but I think it would be a redundant code, this should just compare those two things).

Reply:
Added UTs for the compare method.